src/pkg/db/dependency.go

```diff
@@ -13,19 +13,44 @@ import (
 type Dependency struct {
 	Model
 
-	TaxonomyID  uuid.UUID        `gorm:"default:null"` // Given taxonomy's uncertain presence in YAML, setting TaxonomyID default as NULL accommodates potential absence of taxonomy data.
-	InterfaceID string           `gorm:"unique"`
-	Title       string           `gorm:"default:null"`
-	Description string           `gorm:"default:null"`
-	Inputs      *jsonschema.Node `gorm:"type:json"`
-	Outputs     *jsonschema.Node `gorm:"type:json"`
-	ExtendsID   string           `gorm:"-"` //This is yet to be finalized
-
-	Taxonomy *Taxonomy `gorm:"foreignKey:TaxonomyID"`
+	TaxonomyID  uuid.UUID `gorm:"default:null"` // Given taxonomy's uncertain presence in YAML, setting TaxonomyID default as NULL accommodates potential absence of taxonomy data.
+	InterfaceID string    `gorm:"unique"`
+	Title       string    `gorm:"default:null"`
+	Description string    `gorm:"default:null"`
+	ExtendsID   string    `gorm:"-"` //This is yet to be finalized
+
+	Attributes []DependencyAttribute `gorm:"foreignKey:DependencyID"`
+	Taxonomy   *Taxonomy             `gorm:"foreignKey:TaxonomyID"`
 }
 
 type Dependencies []*Dependency
 
+type DependencyOutput struct {
+	Dependency
+	Inputs  *jsonschema.Node `json:"inputs"`
+	Outputs *jsonschema.Node `json:"outputs"`
+}
+
+type DependencyOutputs []DependencyOutput
+
+func (do DependencyOutputs) ToProto() []*terrariumpb.Dependency {
+	var res []*terrariumpb.Dependency
+	for _, output := range do {
+		res = append(res, output.ToProto())
+	}
+	return res
+}
+
+func (do *DependencyOutput) ToProto() *terrariumpb.Dependency {
+	return &terrariumpb.Dependency{
+		InterfaceId: do.InterfaceID,
+		Title:       do.Title,
+		Description: do.Description,
+		Inputs:      JSONSchemaToProto(do.Inputs),
+		Outputs:     JSONSchemaToProto(do.Outputs),
+	}
+}
+
 // insert a row in DB or in case of conflict in unique fields, update the existing record and set the existing record ID in the given object
 func (db *gDB) CreateDependencyInterface(e *Dependency) (uuid.UUID, error) {
```

Comment:
There is a lot of un-covered code in the dependency.go it seems. Is it expected? The ToProto functions should be easy to test individually.